project/backend/src/test/java/com/google/sps/JsonToRestaurantParserTest.java

```diff
@@ -0,0 +1,112 @@
+  private static final String PLACE_ID = "A12345";
+  private static final String VICINITY = "10 Main Street";
+  private static final Map<String, Double> COORDS = new HashMap<String, Double>() {{
+    put("lat", 10.0);
+    put("lng", 20.0);
+  }};
+
+  @Test
+  public void allRequiredAndOptionalFields() throws JSONException {
+    // JSON object has all required and all optional fields.
+    // We should see a restaurant with all the values in the JSON object.
+    JSONObject geometry = new JSONObject().put("location", COORDS);
+    JSONObject body = new JSONObject()
+      .put("place_id", PLACE_ID)
+      .put("name", RESTAURANT_NAME)
+      .put("vicinity", VICINITY)
+      .put("types", EMPTY_TYPES)
+      .put("geometry", geometry)
+      .put("rating", RATING)
+      .put("user_ratings_total", NUM_RATINGS)
+      .put("price_level", PRICE_LEVEL);
+
+    Restaurant expected = Restaurant.create(
+      PLACE_ID, RESTAURANT_NAME, VICINITY, COORDS, RATING, NUM_RATINGS,
+      PRICE_LEVEL, EMPTY_TYPES);
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void extraneousTypes() throws JSONException {
+    // JSON object has types that should be disregarded.
+    // We should see a restaurant with only the three relevant types.
+    List<String> randomTypes = Arrays.asList(new String[]{"type a", "meal_takeaway", "type b", "meal_delivery", "restaurant"});
+    JSONObject geometry = new JSONObject().put("location", COORDS);
+    JSONObject body = new JSONObject()
+      .put("place_id", PLACE_ID)
+      .put("name", RESTAURANT_NAME)
+      .put("vicinity", VICINITY)
+      .put("types", randomTypes)
+      .put("geometry", geometry);
+
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
+    assertThat(actual.getPlaceTypes()).containsExactlyElementsIn(ALL_TYPES);
+  }
+
+  @Test
+  public void noOptionalFields() throws JSONException {
+    // JSON object has all required fields and no optional fields.
+    // We should see a restaurant that has the required fields and 
+    // default values for optional fields.
+    JSONObject geometry = new JSONObject().put("location", COORDS);
+    JSONObject body = new JSONObject()
+      .put("place_id", PLACE_ID)
+      .put("name", RESTAURANT_NAME)
+      .put("vicinity", VICINITY)
+      .put("types", EMPTY_TYPES)
+      .put("geometry", geometry);
+
+    Restaurant expected = Restaurant.create(
+      PLACE_ID, RESTAURANT_NAME, VICINITY, COORDS, /* avgRating= */ MISSING_FIELD, 
+      /* numRatings= */ MISSING_FIELD, /* priceNum= */ MISSING_FIELD, EMPTY_TYPES);
+    Restaurant actual = JsonToRestaurantParser.toRestaurant(body);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test(expected = JSONException.class)
+  public void noFields() throws JSONException {
+    // JSON object has no fields.
+    // We should see a JSONException thrown.
```

Comment:
nit: I don;t think these comments are needed as the expected= and test method name says exactly this 
(consider comment usefulness by looking at for instance - does it just describe the method name or what the code is doing or do I need a comment to explain why something was done a certain way?)